src/App/Pages/Generator/GeneratorPageViewModel.cs

```diff
@@ -91,6 +95,19 @@ public GeneratorPageViewModel()
             RegenerateUsernameCommand = new AsyncCommand(RegenerateUsernameAsync, onException: ex => OnSubmitException(ex), allowsMultipleExecutions: false);
             ToggleForwardedEmailHiddenValueCommand = new AsyncCommand(ToggleForwardedEmailHiddenValueAsync, onException: ex => OnSubmitException(ex), allowsMultipleExecutions: false);
             CopyCommand = new AsyncCommand(CopyAsync, onException: ex => OnSubmitException(ex), allowsMultipleExecutions: false);
+            CloseCommand = new AsyncCommand(CloseAsync, onException: ex => OnSubmitException(ex), allowsMultipleExecutions: false);
+        }
+
+        public async Task CloseAsync()
+        {
+            if (_iosExtension)
+            {
+                _deviceActionService.CloseExtensionPopUp();
+            }
+            else
+            {
+                await Page.Navigation.PopModalAsync();
+            }
         }
```

Comment:
Methods should be below properties, you can move this to the end of the file if you want